qiskit_ibm/ibm_provider.py

```diff
@@ -27,51 +28,83 @@
 
 from qiskit_ibm.runtime import runtime_job  # pylint: disable=unused-import
-from qiskit_ibm import ibmqfactory  # pylint: disable=unused-import,cyclic-import
-
-from .api.clients import AccountClient
-from .ibmqbackend import IBMQBackend, IBMQSimulator  # pylint: disable=cyclic-import
-from .credentials import Credentials
-from .ibmqbackendservice import IBMQBackendService  # pylint: disable=cyclic-import
+from qiskit_ibm import ibm_account  # pylint: disable=unused-import,cyclic-import
+
+from .api.clients import AuthClient, AccountClient, VersionClient
+from .apiconstants import QISKIT_IBM_API_URL
+from .ibm_backend import IBMBackend, IBMSimulator  # pylint: disable=cyclic-import
+from .credentials import Credentials, discover_credentials
+from .credentials.exceptions import HubGroupProjectInvalidStateError
+from .ibm_backend_service import IBMBackendService  # pylint: disable=cyclic-import
 from .utils.json_decoder import decode_backend_configuration
-from .random.ibmqrandomservice import IBMQRandomService  # pylint: disable=cyclic-import
+from .random.ibm_random_service import IBMRandomService  # pylint: disable=cyclic-import
 from .experiment import IBMExperimentService  # pylint: disable=cyclic-import
 from .runtime.ibm_runtime_service import IBMRuntimeService  # pylint: disable=cyclic-import
-from .exceptions import IBMQNotAuthorizedError, IBMQInputValueError
+from .exceptions import (IBMNotAuthorizedError, IBMInputValueError, IBMAccountCredentialsNotFound,
+                         IBMAccountCredentialsInvalidFormat, IBMAccountCredentialsInvalidToken,
+                         IBMAccountCredentialsInvalidUrl, IBMProviderValueError)
 from .runner_result import RunnerResult  # pylint: disable=cyclic-import
 
 logger = logging.getLogger(__name__)
 
 
-class AccountProvider(Provider):
-    """Provider for a single IBM Quantum Experience account.
+class IBMProvider(Provider):
+    """Provider for a single IBM Quantum account.
 
-    The account provider class provides access to the IBM Quantum Experience
+    This class provides access to the IBM Quantum
     services available to this account.
 
-    You can access a provider by enabling an account with the
-    :meth:`IBMQ.enable_account()<IBMQFactory.enable_account>` method, which
+    You can access the default open provider by instantiating this class
+    and providing the API token.
+
+        from qiskit_ibm import IBMProvider
+        provider = IBMProvider(token=<INSERT_IBM_QUANTUM_TOKEN>)
+
+    To access a different provider, specify the hub, group and project name of the
+    desired provider during instantiation.
+
+        from qiskit_ibm import IBMProvider
+        provider = IBMProvider(token=<INSERT_IBM_QUANTUM_TOKEN>, hub='a', group='b', project='c')
+
+    Instead of passing in the parameters during instantiation, you can also set the environment
+    variables QISKIT_IBM_API_TOKEN, QISKIT_IBM_API_URL, QISKIT_IBM_HUB, QISKIT_IBM_GROUP
```

Comment:
Why are `QISKIT_IBM_HUB` etc parsed?

Reply:
They are read here https://github.com/Qiskit-Partners/qiskit-ibm/blob/main/qiskit_ibm/credentials/environ.py#L21 and I had renamed them in this commit https://github.com/Qiskit-Partners/qiskit-ibm/commit/ea9a3a7c7114a9a7ce7ff014e50a837143e61c63